addons/web/static/src/core/py_js/py_date.js

```diff
@@ -580,24 +639,70 @@ export class PyRelativeDelta {
      * @returns {PyDateTime}
      */
     static substract(date, delta) {
+        date = delta.applyAbsolute(date);
+        const leapDays = delta.getLeapDays(date);
+
         const s = tmxxx(
             date.year,
             date.month,
-            date.day - delta.day,
+            date.day - delta.day + leapDays,
             -delta.hour,
             -delta.minute,
             -delta.second
         );
+
+        if (delta.weekday !== null) {
+            const sprime = delta.applyWeekday(s);
+            Object.assign(s, sprime);
+        }
+
         return new PyDateTime(s.year, s.month, s.day, s.hour, s.minute, s.second, 0);
     }
 
+    applyAbsolute(oldDdate) {
+        const date = new PyDateTime();
+        Object.assign(date, oldDdate);
```

Comment:
très moche ça :-)

Reply:
hmmm, this is defensive, my thinking is that since we are evaluating a short, controlled string (namely a domain) the objects won't be re-used later in the python code.
So: we can alter objects in place, and we can be a bit more approximative on types.

what do you think ?

Reply:
later on  date.hour + delta.hour becomes NaN if you have a date and a relativedelta with hours=3 for example. In that condition, there is no crash but the relativedelta is not correctly applied